### PR TITLE
Kubemark e2e tests should exit 0 for test failures.

### LINKED ISSUE #26318: Kubemark: Use unstable state, don't fail completely
https://github.com/kubernetes/kubernetes/blob/master/hack/jenkins/e2e-runner.sh#L367

If we fail completely, it blocks the submit queue.

@wojtek-t or @gmarek can one of you fix this ~asap? This is blocking the merge queue unnecessarily. See: http://kubekins.dls.corp.google.com/view/Critical%20Builds/job/kubernetes-kubemark-500-gce/3375/

### PR DESCRIPTION
**Pull request by pwittrock**
Fixed #26318

### CODE REVIEW

**Comment by pwittrock** [comment]
Will re-assign once all tests pass

File: hack/jenkins/e2e-runner.sh [hunk 1]
```diff
@@ -151,7 +158,6 @@ function dump_cluster_logs_and_exit() {
       # as possible and don't even know if we brought the master up.)
       go run ./hack/e2e.go ${E2E_OPT:-} -v --down || true
```
Comment (ixdy) [inline hack/jenkins/e2e-runner.sh]: maybe move this section to the _exit() function instead?

Comment (pwittrock) [inline hack/jenkins/e2e-runner.sh]: Good catch.  Done

**Comment by ixdy** [comment]
LGTM

**Comment by pwittrock** [comment]
@ixdy Thanks

**Comment by wojtek-t** [comment]
LGTM - thanks @pwittrock !

**Pull request merged**